src/hooks/useAsset.js

```diff
@@ -58,28 +57,14 @@ export default function useAsset(asset) {
   const genericAssets = useSelector(
     ({ data: { genericAssets } }) => genericAssets
   );
-  const uniswapAssetsInWallet = useUniswapAssetsInWallet();
 
   return useMemo(() => {
     if (!asset) return null;
 
     let matched = null;
     if (asset.type === AssetTypes.token) {
-      const uniswapAsset = find(
-        uniswapAssetsInWallet,
-        matchesProperty('address', asset.mainnet_address || asset.address)
-      );
+      matched = find(allAssets, matchesProperty('uniqueId', asset.uniqueId));
 
-      matched = uniswapAsset
-        ? uniswapAsset
-        : find(
-            allAssets,
-            matchesProperty('address', asset.mainnet_address || asset.address)
```

Comment:
we were also doing 2 redundant finds in `allAssets` - first with address and then with uniqueId. Now I just have it using uniqueId.